client/core/wallet.go

```diff
@@ -141,18 +141,19 @@ func (w *xcWallet) state() *WalletState {
 	defer w.mtx.RUnlock()
 	winfo := w.Info()
 	return &WalletState{
-		Symbol:       unbip(w.AssetID),
-		AssetID:      w.AssetID,
-		Version:      winfo.Version,
-		Open:         len(w.encPass) == 0 || len(w.pw) > 0,
-		Running:      w.connector.On(),
-		Balance:      w.balance,
-		Address:      w.address,
-		Units:        winfo.UnitInfo.AtomicUnit,
-		Encrypted:    len(w.encPass) > 0,
-		Synced:       w.synced,
-		SyncProgress: w.syncProgress,
-		WalletType:   w.walletType,
+		Symbol:        unbip(w.AssetID),
+		AssetID:       w.AssetID,
+		Version:       winfo.Version,
+		Open:          len(w.encPass) == 0 || len(w.pw) > 0,
+		Running:       w.connector.On(),
+		Balance:       w.balance,
+		Address:       w.address,
+		Units:         winfo.UnitInfo.AtomicUnit,
+		Encrypted:     len(w.encPass) > 0,
+		Synced:        w.synced,
+		SyncProgress:  w.syncProgress,
+		WalletType:    w.walletType,
+		RegisterTxFee: w.EstimateRegistrationTxFee(),
```

Comment:
I think the logical places it would be is `core.Exchange` along side the `RegFees` map/object.
This would put it in the `getdexconfig` result (our app doesn't use it, but other consumers do), as well as in the `discoveraccount` result.

BUT, I would only go for that if this does not involve making fee rate requests, or if the the request is only made and an optional field set in certain conditions (caller is GetDexConfig or DiscoverAccount in the not found case).

Reply:
OK well you wouldn't necessarily have the necessary wallets at that point.

Reply:
To selectively include the fee rates in `core.Exchange.Regfees`, I'm thinking this:

```diff
diff --git a/client/core/types.go b/client/core/types.go
index 41845440..133333ae 100644
--- a/client/core/types.go
+++ b/client/core/types.go
@@ -460,9 +460,10 @@ func (m *Market) marketName() string {
 // FeeAsset has an analogous msgjson type for server providing supported
 // registration fee assets.
 type FeeAsset struct {
-       ID    uint32 `json:"id"`
-       Confs uint32 `json:"confs"`
-       Amt   uint64 `json:"amount"`
+       ID      uint32  `json:"id"`
+       Confs   uint32  `json:"confs"`
+       Amt     uint64  `json:"amount"`
+       FeeRate *uint64 `json:"feeRate,omitempty"`
 }
```

Then just for the methods that use a temp dex connection:

```diff
diff --git a/client/core/core.go b/client/core/core.go
index 77fd2f6c..cdcad268 100644
--- a/client/core/core.go
+++ b/client/core/core.go
@@ -336,6 +336,16 @@ func (dc *dexConnection) exchangeInfo() *Exchange {
        }
 }
 
+func (dc *dexConnection) exchangeInfoWithRegFeeRates() *Exchange {
+       xc := dc.exchangeInfo()
+       for _, fa := range xc.RegFees {
+               if rate := dc.fetchFeeRate(fa.ID); rate != 0 {
+                       fa.FeeRate = &rate
+               }
+       }
+       return xc
+}
+
 // hasActiveAssetOrders checks whether there are any active orders or negotiating
 // matches for the specified asset.
 func (dc *dexConnection) hasActiveAssetOrders(assetID uint32) bool {
@@ -2454,7 +2464,7 @@ func (c *Core) GetDEXConfig(dexAddr string, certI interface{}) (*Exchange, error
 
        // Since connectDEX succeeded, we have the server config. exchangeInfo is
        // guaranteed to return an *Exchange with full asset and market info.
-       return dc.exchangeInfo(), nil
+       return dc.exchangeInfoWithRegFeeRates(), nil
 }
 
 // discoverAccount attempts to identify existing accounts at the connected DEX.
@@ -2598,7 +2608,7 @@ func (c *Core) DiscoverAccount(dexAddr string, appPW []byte, certI interface{})
        // Older DEX server. We won't allow registering without an HD account key,
        // but discovery can conclude we do not have an HD account with this DEX.
        if dc.acct.dexPubKey == nil {
-               return dc.exchangeInfo(), false, nil
+               return dc.exchangeInfoWithRegFeeRates(), false, nil
        }
 
        // Setup our account keys and attempt to authorize with the DEX.
@@ -2607,7 +2617,7 @@ func (c *Core) DiscoverAccount(dexAddr string, appPW []byte, certI interface{})
                return nil, false, err
        }
        if !paid {
-               return dc.exchangeInfo(), false, nil // all good, just go register now
+               return dc.exchangeInfoWithRegFeeRates(), false, nil // all good, just go register now
        }
```